graphql-kotlin-schema-generator/src/test/kotlin/com/expedia/graphql/hooks/SchemaGeneratorHooksTest.kt

```diff
@@ -120,8 +98,10 @@ class SchemaGeneratorHooksTest {
             queries = listOf(TopLevelObject(TestQuery())),
             config = getTestSchemaConfigWithHooks(hooks)
         )
-        assertEquals(SomeData::class.createType(), hooks.lastSeenType)
-        assertEquals("SomeData!", hooks.lastSeenGeneratedType?.deepName)
+        assertTrue(hooks.seenTypes.contains(RandomData::class.createType()))
+        assertTrue(hooks.seenTypes.contains(SomeData::class.createType()))
+        // TODO bug: didGenerateGraphQLType hook is not applied on the object types build from the interface
```

Comment:
TODO intended here?

Reply:
yes -> it is a bug in how we apply the hooks... :(